### ISSUE TITLE
Error: headless task failed, on imagebuild

### ISSUE THREAD

**Issue by axonasif** [gitpod-io/gitpod#9875]
### Bug description

I was trying to open this repo https://gitpod.io/#https://github.com/axonasif/wrover-experimental/tree/wokwi-test after adding a custom dockerfile
<img width="1060" alt="Screenshot 2022-05-09 at 9 47 51 PM" src="https://user-images.githubusercontent.com/39482679/167447779-5597c636-d1ca-4219-af77-b61ec55d81b4.png">

### Steps to reproduce

Open https://github.com/axonasif/wrover-experimental/tree/wokwi-test on Gitpod.

### Workspace affected

axonasif-wroverexperime-u9ik2sre8pl

### Example repository

https://github.com/axonasif/wrover-experimental/tree/wokwi-test

### Anything else?

A similar report on Discord today: https://discord.com/channels/816244985187008514/972590280446476308/973228246407733288

**Comment by ChamMach** [comment]
+1, same for me 🤔

**Comment by felipealencar** [comment]
Same problem here.

**Comment by ChamMach** [comment]
Very surprised that there aren't more people complaining about it. Although it's about viewing the outputs/logs, for me, it becomes *complicated* not knowing exactly where the problem is

**Comment by felipealencar** [comment]
Hello everyone, Idk who still is facing this issue, but I could see detailed information about my error by:
1. Selecting the option to run a default workspace;
2. Running my custom `.gitpod.Dockerfile` in a test:
`docker build -f .gitpod.Dockerfile -t gitpod-dockerfile-test .`
3. Finally, I could see the error messages related to the problems of my Dockerfile.

I hope it helps someone.

**Comment by geropl** [comment]
Scheduled this, as the user experience feels really broken :confused: .

@felipealencar Thank you for sharing the workaround! This is indeed what's happening: The `headless task failed: exit status 1` error is always emitted if there's some user-error with the image build.
That could be anything from error pulling the image, to a failing Docker command inside the docker file.

**Comment by ChamMach** [comment]
Yep, thanks for the workaround! I realized it too by simply building the *problematic Dockerfile* on another gitpod.

Thanks to the team for taking this issue into account too! 🙏

**Comment by jankeromnes** [comment]
Maybe this error is related? (internal) https://console.cloud.google.com/errors/detail/CNif-a-gteWfNw;service=server;version=main.3268;time=PT1H?project=gitpod-191109

**Comment by csweichel** [comment]
@jankeromnes @meysholdt looks like a certificate issue with the new application clusters:
```
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","buildID":"ca0da695-1b59-4827-9adf-f2054be7be7d","error":"cannot connect to supervisor:\n    github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator.findTaskLogURL\n        github.com/gitpod-io/gitpod/image-builder/pkg/orchestrator/monitor.go:396\n  - Get \"https://ca0da695-1b59-4827-9adf-f2054be7be7d.ws.gitpod.io/_supervisor/v1/status/tasks\": GET https://ca0da695-1b59-4827-9adf-f2054be7be7d.ws.gitpod.io/_supervisor/v1/status/tasks giving up after 11 attempt(s): Get \"https://ca0da695-1b59-4827-9adf-f2054be7be7d.ws.gitpod.io/_supervisor/v1/status/tasks\": x509: certificate is valid for gitpod.io, not ca0da695-1b59-4827-9adf-f2054be7be7d.ws.gitpod.io","level":"warning","message":"headless log listener failed","serviceContext":{"service":"image-builder-mk3","version":"commit-40cfaec3f92cc6cadc42dfd5e6028b8c3049a16c"},"severity":"WARNING","time":"2022-05-17T22:13:32Z"}
```

**Comment by meysholdt** [comment]
In case the above error message is from production, then the request
```Get \"https://ca0da695-1b59-4827-9adf-f2054be7be7d.ws.gitpod.io/_supervisor/v1/status/tasks```
looks wrong because we don't have a ws-cluster named `ws` in prod. 
Requesting the wrong URL would explain why the domains from the certs don't match. 

There was a change in certs indeed (https://github.com/gitpod-io/ops/pull/2228), but I'm not sure it's related. CC @easyCZ .

**Comment by andrew-farries** [comment]
Working on the assumption that it's the workspace URL that's wrong and not the certificates, these are set here:

https://github.com/gitpod-io/gitpod/blob/faa6b3096c44aeffae9c1a7b0560091bed69d816/install/installer/pkg/components/ws-manager/configmap.go#L140-L141

so if the hardcoded `ws` component in the string is incorrect, then it looks like we missed adding some configuration to the installer to allow this to vary.

But if the URL is wrong, why do we see an error about certs rather than have the request time out?

**Comment by andrew-farries** [comment]
This is how we used to construct these URLs in the helm chart:

https://github.com/gitpod-io/gitpod/blob/2f988c93499926bd34d6728defbd9d9c0964797e/chart/templates/ws-manager-configmap.yaml#L74-L75

so it does look as though we need to set them via the installer in a similar way...

**Comment by mteam88** [comment]
Really struggling without this!

**Comment by andrew-farries** [comment]
Re-opening this as #10127 had to be reverted because it broke preview environments (see [internal discussion](https://gitpod.slack.com/archives/C032A46PWR0/p1652996871215559)).

The installer (which we also use to deploy to preview environments) defaults to setting:

https://github.com/gitpod-io/gitpod/blob/b765e67295b42834445ae2078b8fa8452fe01467/install/installer/pkg/config/v1/config.go#L50

Which breaks when used as part of the URL template strings in a preview env installation (we don't have a cert for that domain, and I'm not sure image build logs would be available there anyway).

Maybe @geropl can provide some context about why we need that default when he is back.

**Comment by WesleyMConner** [comment]
Can you add a little color on how best to debug?

Historically, I have used a three step process.

1. Performed a docker build of the prospective `.gitpod.Dockerfile` (in Gitpod itself).

----
docker build -f .gitpod.Dockerfile -t gitpod-dockerfile-test .
----

2. If #1 ran without issues, launch  a local container (still in Gitpod) and do some spot checks.

----
docker run --rm -it gitpod-dockerfile-test /bin/bash
----

3. Commit the changes and fire up a new Gitpod instance (with the revised `.gitpod.Dockerfile`) and check for expected behavior.

Lately, I get through steps 2 and 3 with no obvious issues. Then, run into the "Error: headless task failed" message. Is there a step I can/should be doing and am missing OR is there something happening here that isn't easily smoked out? 

Thanks,
Wes
